waitress/proxy_headers.py

```diff
@@ -0,0 +1,335 @@
+from collections import namedtuple
+
+from .utilities import logger, undquote, BadRequest
+
+
+PROXY_HEADERS = frozenset({
+    'HTTP_X_FORWARDED_FOR',
+    'HTTP_X_FORWARDED_HOST',
+    'HTTP_X_FORWARDED_PROTO',
+    'HTTP_X_FORWARDED_PORT',
+    'HTTP_X_FORWARDED_BY',
+    'HTTP_FORWARDED',
+})
+
+Forwarded = namedtuple('Forwarded', ['by', 'for_', 'host', 'proto'])
+
+
+class MalformedProxyHeader(Exception):
+    def __init__(self, header, reason):
+        self.header = header
+        self.reason = reason
+        Exception.__init__(self, header, reason)
```

Comment:
Not a huge fan of this style of calling the super class init, I would prefer to see `super`.

Reply:
ok